src/Miscellaneous/Mis_Matching.cpp

```diff
@@ -59,13 +59,14 @@ int Mis_Matching_char( const int N, const T Array[], const int M, const T Key[],
 //-------------------------------------------------------------------------------------------------------
 // for int Match[]
 //-------------------------------------------------------------------------------------------------------
-template <typename T>
-int Mis_Matching_int( const int N, const T Array[], const int M, const T Key[], int Match[] )
+template <typename U, typename T>
+int Mis_Matching_int( const U N, const T Array[], const U M, const T Key[], U Match[] )
```

Comment:
Yes, I think we should redefine the type for `NMatch`, since its maximum can be as large as `M`, so might exceed  `__INT_MAX__`, though currently this make no influence since the returned value of `Mis_Matching_int` seems to be used at nowhere in `GAMER`.